pkg/network/ebpf/c/protocols/tls/https.h

```diff
@@ -48,25 +52,68 @@ static __always_inline void classify_decrypted_payload(conn_tuple_t *t, void *bu
     set_protocol(stack, proto);
 }
 
-static __always_inline void http_process(http_transaction_t *http_stack, skb_info_t *skb_info, __u64 tags);
+static __always_inline void tls_process(conn_tuple_t *t, void *buffer_ptr, size_t len, __u64 tags) {
+    protocol_stack_t *stack = get_protocol_stack(t);
+    if (!stack) {
+        return;
+    }
 
-static __always_inline void https_process(conn_tuple_t *t, void *buffer, size_t len, __u64 tags) {
-    http_transaction_t http;
-    bpf_memset(&http, 0, sizeof(http));
-    bpf_memcpy(&http.tup, t, sizeof(conn_tuple_t));
-    read_into_buffer(http.request_fragment, buffer, len);
-    http_process(&http, NULL, tags);
-    classify_decrypted_payload(&http.tup, http.request_fragment, len);
+    const __u32 zero = 0;
+    char *request_fragment = bpf_map_lookup_elem(&tls_classification_heap, &zero);
+    if (request_fragment == NULL) {
+        return;
+    }
+    read_into_buffer_classification(request_fragment, buffer_ptr, len);
+
+    classify_decrypted_payload(stack, t, request_fragment, len);
+
+    protocol_t protocol = get_protocol_from_stack(stack, LAYER_APPLICATION);
+    switch (protocol) {
+    case PROTOCOL_HTTP: {
+        http_transaction_t *http = bpf_map_lookup_elem(&http_transaction_heap, &zero);
+        if (http == NULL) {
+            return;
```

Comment:
extreme edge case right? should we log here? or even better I think we can add that map operation to ebpf_telemetry, as we shouldn't fail in this case)

Reply:
I'm uncertain about map telemetry, that code piece is quite heavy
regarding the edge case, while it can happen theoretically (if giving wrong index) - it can never happen (we have an array with only 1 entry, and we're accessing the element in the first location)

Reply:
we don't use that macro in the project